internal/rootcoord/meta_table.go

```diff
@@ -356,6 +353,11 @@ func (mt *MetaTable) GetCollectionByName(ctx context.Context, collectionName str
 	if err != nil {
 		return nil, err
 	}
+
+	if coll != nil && allowUnavailable {
```

Comment:
Please cover this case if you necessarily need this feature. And do not use `Clone` here, it makes no sense.